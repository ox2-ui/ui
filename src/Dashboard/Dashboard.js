import React, { PropTypes } from 'react';
import Scroller from '@ox2/scroller/Scroller';
import Button from '@ox2/button/Button';

/**
 * Dashboard Component
 */
const Dashboard = ({ onDismiss }) => (
  <Scroller className="padding-b:40">
    <Button onClick={() => onDismiss()} btn="raised action">Done</Button>
    <div>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Itaque, sequi? Consequuntur inventore nulla ratione vitae obcaecati voluptatem animi molestiae ipsam et adipisci illo reiciendis saepe doloremque, quae perspiciatis iure vero.</div>
  </Scroller>
);

Dashboard.propTypes = {
  onDismiss: PropTypes.func.isRequired,
};

export default Dashboard;
