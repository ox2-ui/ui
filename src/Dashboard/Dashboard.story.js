import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Dashboard from './Dashboard';

const handler = () => console.log('🍇'); // eslint-disable-line no-console

storiesOf('Dashboard', module)
  .add('default', () => (
    <Dashboard onDismiss={handler} />
  ));
