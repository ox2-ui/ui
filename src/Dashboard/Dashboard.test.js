/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Dashboard from './Dashboard';

const testHandler = () => true;

it('renders correctly', () => {
  const wrapper = mount(
    <Dashboard onDismiss={testHandler} />
  );

  expect(wrapper).toMatchSnapshot();
});
