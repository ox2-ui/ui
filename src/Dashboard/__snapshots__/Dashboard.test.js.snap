// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Dashboard
  onDismiss={[Function]}
>
  <Scroller
    className="padding-b:40"
  >
    <div
      className="padding-b:40"
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitOverflowScrolling": "touch",
          "backgroundColor": "transparent",
          "bottom": "0",
          "left": "0",
          "overflowX": "hidden",
          "overflowY": "auto",
          "paddingBottom": "40px",
          "position": "absolute",
          "right": "0",
          "top": "0",
        }
      }
    >
      <Button
        btn="raised action"
        onClick={[Function]}
      >
        <button
          data-btn="raised action"
          onClick={[Function]}
        >
          Done
        </button>
      </Button>
      <div>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Itaque, sequi? Consequuntur inventore nulla ratione vitae obcaecati voluptatem animi molestiae ipsam et adipisci illo reiciendis saepe doloremque, quae perspiciatis iure vero.
      </div>
    </div>
  </Scroller>
</Dashboard>
`;
